src/formatDate.js

```diff
@@ -8,7 +8,28 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const dateParts = date.split(fromSeparator);
+
+  const dateComponents = {};
+
+  fromFormat.slice(0, 3).forEach((part, index) => {
+    dateComponents[part] = dateParts[index];
+  });
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    dateComponents['YYYY'] =
+      dateComponents['YY'] < 30
+        ? `20${dateComponents['YY']}`
+        : `19${dateComponents['YY']}`;
+  } else if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    dateComponents['YY'] = dateComponents['YYYY'].slice(-2);
+  }
+
+  const newDateParts = toFormat.slice(0, 3).map((part) => dateComponents[part]);
```

Comment:
When mapping over the `toFormat` array to create `newDateParts`, you've assumed that all parts ('YYYY', 'MM', 'DD') exist in `dateComponents`. This assumption holds because of the task's constraints, but consider adding a comment above this line explaining that due to task constraints, no validation is needed here.